src/components/Share.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import {shareGoogleDrive} from '../redux/actionCreators';
+import {bindActionCreators} from 'redux';
+import {styles} from './styles';
+import Button from './Button';
+import Spinner from './Spinner';
+import * as selectors from '../redux/selectors';
+
+export class Share extends React.Component {
+  render() {
+    return (
+      <span>
+        {this.props.hasSavedDrive && !this.props.isSharingDrive &&
+          <Button kind="toolbar"
+                  onClick={() => this.props.shareGoogleDrive()}>
+            Share
+          </Button>
+        }
+        {this.props.hasSavedDrive && this.props.isSharingDrive &&
+          <Button kind="toolbar">
+            <Spinner style={styles.spinners.toolbar}/>
+            Sharing...
+          </Button>
+        }
+      </span>
```

Comment:
Instead of checking `this.props.hasSavedDrive` twice, I think it would be better to do an early return. i.e:

```javascript
render() {
  if (!this.props.hasSavedDrive) {
    return null
  }
  ...
```